Add multiple image upload and preview to book listing form

Emjodez opened the pull request:
- Updated the file input to accept multiple images
- Implemented client-side preview container for uploaded images

Emjodez committed:
Add support for uploading multiple images of the book to the Add page.

Emjodez committed:
Adjust size of second image when there are exactly 2 images

Emjodez committed:
Update dotnet-tools.json

Emjodez committed:
Remove unneeded files

Emjodez committed:
Remove unneeded files.

Emjodez committed:
Remove unneeded files

Emjodez committed:
Add missing files.

Emjodez committed:
Preserve original file names.

Emjodez committed:
Merge branch 'booksAPI' into feature/multiple-image-upload-frontend

MrFranQx commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,182 @@
-                } else {
-                    if (height > MAX_HEIGHT) {
+    const imageErrorSpan = input.closest("section").querySelector("small span");
+    const files = Array.from(input.files).slice(0, 6);
```
Personally, I would explicitly reject uploading more than 6 files (or whatever the number) with an error message rather than silently truncate the array.

MrFranQx commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,182 @@
+                            imageElement.id = `Image${index + 1}`;
+                            imageElement.classList.add("book-image-preview", "active");
+                            preview.appendChild(imageElement);
+                            imageErrorSpan.style.display = "none";
```
Why is this here I have no idea. I see it's been here before, but I don't think it's needed; no other validation logic sets the display to `none`.

MrFranQx commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,182 @@
+        .catch(err => {
+            console.error("Image upload failed:", err);
+            imageErrorSpan.textContent = "Błąd przetwarzania obrazu.";
+            imageErrorSpan.style.display = "block";
```
Remove this as well.
I think we would need to somehow hook up the validation logic to `aspnet-validation`, but I don't know how yet.

MrFranQx commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,182 @@
+        });
+}
+
+function applyStyling(imageAmount, preview) {
```
That much styling shouldn't be done dynamically. Create regular CSS classes in `_add.scss` and add them dynamically. Also, I see you're using Flexbox, but the layout is two-dimensional; try to do it with Grid instead.

MrFranQx commented on Booker/Pages/Add.cshtml:
Don't know how this happened, but this file has a double BOM and the project does not compile.

MrFranQx reviewed (changes requested):
Looks promising, but needs a few changes, particularly with the styling.

Emjodez committed:
Display error message when more than 6 images are uploaded.
Move image styling from JavaScript to dedicated _add.scss file using CSS classes.
Replace unnecessary flexbox containers with CSS grid.

Emjodez committed:
Merge branch.

Emjodez committed:
Fix formatting, modify _add.scss.

theluckyprogrammer commented:
> Looks promising, but needs a few changes, particularly with the styling.

Could you briefly elaborate on this point?

theluckyprogrammer commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,139 @@
+    const preview = input.closest("section").querySelector(".image-preview-container");
+    preview.innerHTML = "";
+    const imageErrorSpan = input.closest("section").querySelector("#imageErrorMsg");
+    if (input.files.length > 6) {
```
it lools like a good place to use elvis operator '?' ? (;D)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator

Emjodez replied:
I don't think it's possible to use a ternary conditional here because the function needs to end early if there are more than 6 images. Return statements can't be used inside ternary expressions.

theluckyprogrammer commented on Booker/wwwroot/js/site.js:
```diff
@@ -4,65 +4,139 @@
+
+    const processingPromises = files.map((file, index) => {
+        return new Promise((resolve, reject) => {
+            console.log(index);
```
remove console.log

theluckyprogrammer reviewed (commented):
Welcome with the first PR in our solution. :) 
Good to have this feature in place.

Emjodez committed:
Remove redundant console.log statements.

Emjodez committed:
Fix visual inconsistencies.

Emjodez committed:
Merge remote-tracking branch 'origin/booksAPI' into feature/multiple-image-upload1

Emjodez committed:
- Updated backend to handle multiple uploaded images per book without overwriting old single-image behavior
- Modified Index page and BookTile to display only the first image for consistency
- Updated site.js showSummary to display the first image
- Ensures backward compatibility with existing single-image books

Emjodez committed:
Revert changes.

Emjodez committed:
Revert accidental changes.

MrFranQx commented on Booker/Pages/Add.cshtml:
```diff
@@ -6,15 +6,17 @@
+            <input type="file" asp-for="Input.Images" accept="image/*" multiple hx-on:change="handleImageUpload(this)">
+            <div class="image-preview-container">
+            </div>
+            <small><span asp-validation-for="Input.Images"></span></small>
```
In Pico, the `<small>` element has to be right after the related input, and there needs to be only one.
It's handled by this CSS rule:
```css
:where(input,select,textarea,fieldset,.grid)+small
```

MrFranQx commented on Booker/Booker.csproj:
```diff
@@ -37,4 +37,8 @@
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="wwwroot\uploads\" />
```
What's this for?

MrFranQx commented on Booker/Pages/Edit.cshtml:
```diff
@@ -26,9 +26,9 @@
-                    <input type="file" asp-for="Input.Image" accept="image/*" hx-on:change="handleImageUpload(this)">
-                    <small><span asp-validation-for="Input.Image"></span></small>
+                    <label asp-for="Input.Images">Zmień zdjęcie swojej książki (opcjonalnie):</label>
+                    <input type="file" asp-for="Input.Images" accept="image/*" hx-on:change="handleImageUpload(this)">
```
`multiple` is needed here

MrFranQx replied:
And also the logic to preview multiple photos

MrFranQx commented on Booker/Pages/Book.cshtml:
```diff
@@ -3,13 +3,34 @@
 <article class="book-display">
+
+  
     <section class="image">
```
<img width="2096" height="1212" alt="image" src="https://github.com/user-attachments/assets/305aff12-9999-4e92-b0d6-ad38d2065b23" />
Whoops

Emjodez replied:
There were some issues with this branch and I'll upload both backend and frontend in a different pull request this week.  This version is not complete yet. (to all comments)

MrFranQx reviewed (changes requested):
Backend looks ok, but frontend has a few bugs and missing features. Also, I haven't checked the JS and CSS yet.

MrFranQx commented:
Will be redone on another branch
